include/util/deallocating_vector.hpp

```diff
@@ -239,6 +239,19 @@ class DeallocatingVector
         bucket_list.emplace_back(new ElementT[ELEMENTS_PER_BLOCK]);
     }
 
+    DeallocatingVector(DeallocatingVector &&other)
+    {
+        bucket_list = std::move(other.bucket_list);
+        current_size = std::move(other.current_size);
+    }
+
+    DeallocatingVector &operator=(DeallocatingVector &&other)
+    {
+        bucket_list = std::move(other.bucket_list);
+        current_size = std::move(other.current_size);
+        return *this;
+    }
```

Comment:
I don't quite understand what you mean. The `DeallocationVector` had an implicitly deleted move constructor.

I had the problem that if you pass a `DeallocationVector` by value, it generated double free errors: If you copy `bucket_list` which is a vector of pointer, there was no deep-copy. The bucket's where owned by two `DeallocationVector` instances and caused problems.

I fixed this by adding a move constructor that safely allows to transfer ownership of buckets. The copy constructor should be deleted though, I don't think there is much sense in allowing deep copies here.